jetty-server/src/test/java/org/eclipse/jetty/server/InclusiveByteRangeTest.java

```diff
@@ -124,48 +138,201 @@ public void testMultipleAbsoluteRanges()
+        rangeString = "bytes=20-30,5-15,0-5,25-35";
 
+        List<InclusiveByteRange> ranges = parseRanges(size,rangeString);
+        assertEquals("Satisfiable Ranges of [" + rangeString + "] count",2,ranges.size());
+        Iterator<InclusiveByteRange> inclusiveByteRangeIterator = ranges.iterator();
+        assertRange("Range [" + rangeString + "]",20,35,size,inclusiveByteRangeIterator.next());
+        assertRange("Range [" + rangeString + "]",0,15,size,inclusiveByteRangeIterator.next());
+    }
+    
+    @Test
+    public void testMultipleOverlappingRangesOrderedSplit()
+    {
+        int size = 200;
+        String rangeString;
+        rangeString = "bytes=20-30,5-15,0-5,25-35";
+        List<InclusiveByteRange> ranges = parseRanges(size,"bytes=20-30","bytes=5-15","bytes=0-5,25-35");
+        
+        assertEquals("Satisfiable Ranges of [" + rangeString + "] count",2,ranges.size());
+        Iterator<InclusiveByteRange> inclusiveByteRangeIterator = ranges.iterator();
+        assertRange("Range [" + rangeString + "]",20,35,size,inclusiveByteRangeIterator.next());
+        assertRange("Range [" + rangeString + "]",0,15,size,inclusiveByteRangeIterator.next());
+    }
+    
+    @Test
+    public void testNasty()
+    {
+        int size = 200;
+        String rangeString;
+
+        rangeString = "bytes=90-100, 10-20, 30-40, -161";
+        List<InclusiveByteRange> ranges = parseRanges(size,rangeString);
+        
+        assertEquals("Satisfiable Ranges of [" + rangeString + "] count",2,ranges.size());
+        Iterator<InclusiveByteRange> inclusiveByteRangeIterator = ranges.iterator();
+        assertRange("Range [" + rangeString + "]",30,199,size,inclusiveByteRangeIterator.next());
+        assertRange("Range [" + rangeString + "]",10,20,size,inclusiveByteRangeIterator.next());
+    }
+    
+    @Test
+    public void testRange_OpenEnded()
+    {
+        assertSimpleRange(50, 499, "bytes=50-", 500);
+    }
+    
     @Test
     public void testSimpleRange()
     {
         assertSimpleRange(5,10,"bytes=5-10",200);
         assertSimpleRange(195,199,"bytes=-5",200);
         assertSimpleRange(50,119,"bytes=50-150",120);
         assertSimpleRange(50,119,"bytes=50-",120);
+        
+        assertSimpleRange(1,50,"bytes= 1 - 50",120);
+    }
+
+    // TODO: evaluate this vs assertInvalidRange() above, which behavior is correct? null? or empty list?
+    private void assertBadRangeList(int size, String badRange)
+    {
+        Vector<String> strings = new Vector<>();
+        strings.add(badRange);
+    
+        List<InclusiveByteRange> ranges = InclusiveByteRange.satisfiableRanges(strings.elements(),size);
+        // if one part is bad, the entire set of ranges should be treated as bad, per RFC7233
+        assertThat("Should have no ranges", ranges, is(nullValue()));
+    }
+    
+    @Test
+    public void testBadRange_SetPartiallyBad()
+    {
+        assertBadRangeList(500, "bytes=1-50,1-b,a-50");
```

Comment:
This is an example of a partially satisfiable range (1-50) with some bad ranges.
This is the kind of request that RFC7233 seems to indicate should result in a 400 Bad Request.